wremnants/CardTool.py

```diff
@@ -27,25 +28,70 @@ def __init__(self, cardName="card.txt"):
         self.addMirrorForSyst = {}
         self.channels = ["plus", "minus"]
         self.cardContent = {}
-        self.cardGroups = ""
+        self.cardGroups = {}
         self.nominalTemplate = ""
         self.spacing = 28
-        self.fakeName = "Fake"
+        self.fakeName = "Fake" # but better to set it explicitly
         self.dataName = "Data"
         self.nominalName = "nominal"
         self.datagroups = None
         self.unconstrainedProcesses = None
+        self.noStatUncProcesses = None
         self.buildHistNameFunc = None
         self.histName = "x"
         self.pseudoData = None
+        self.excludeSyst = None
+        self.keepSyst = None # to override previous one with exceptions for special cases
         #self.loadArgs = {"operation" : "self.loadProcesses (reading hists from file)"}
+        self.keepOtherChargeSyst = True
+        self.chargeIdDict = {"minus" : {"val" : -1, "id" : "q0", "badId" : None},
+                             "plus"  : {"val" : 1., "id" : "q1", "badId" : None}
+                             }
+        
+    def setSkipOtherChargeSyst(self):
+        self.keepOtherChargeSyst = False
+        self.chargeIdDict["plus"]["badId"] = "q0"
+        self.chargeIdDict["minus"]["badId"] = "q1"
+
+    def setProcsNoStatUnc(self, procs, resetList=True):
+        if self.noStatUncProcesses == None or resetList:
+            self.noStatUncProcesses = []
+        if isinstance(procs, str):
+            self.noStatUncProcesses.append(procs)
+        elif isinstance(procs, list):
+            self.noStatUncProcesses.extend(procs)
+        else:
+            raise ValueError("In setNoStatUncForProcs(): expecting string or list argument")            
 
+    def getProcsNoStatUnc(self):
+        return self.noStatUncProcesses
+        
+    ## Functions to customize systs to be added in card, mainly for tests
+    def setCustomSystForCard(self, exclude=None, keep=None):
+        if exclude: self.excludeSyst = re.compile(exclude)
+        if keep:    self.keepSyst    = re.compile(keep)
+        
+    def isExcludedNuisance(self, name):
+        # note, re.match search for a match from the beginning, so if x="test" x.match("mytestPDF1") will NOT match 
+        # might use re.search instead to be able to match from anywhere inside the name
+        if self.excludeSyst != None and self.excludeSyst.match(name):
+            if self.keepSyst != None and self.keepSyst.match(name):
+                return False
+            else:
+                logging.info(f">>>>> Excluding nuisance: {name}")
```

Comment:
Using >>>> is kind of annoying for fixing merge conflicts in git...

Reply:
sorry, it was just an older thing, I will edit